compiler/rustc_ast/src/ast.rs

```diff
@@ -1469,6 +1470,9 @@ pub enum ExprKind {
     /// Output of the `asm!()` macro.
     InlineAsm(P<InlineAsm>),
 
+    /// Output of the `offset_of!()` macro.
+    OffsetOf(P<Ty>, Vec<Ident>),
```

Comment:
I think the only thing missing is to change the `Vec<Ident>` to a `P<[Ident]>` here.

Reply:
Btw. this a pretty trivial change; you can either use `P::from_vec(_)` or `collect()` into it directly (which will call `from_vec` behind the scenes)